autogluon/utils/tabular/ml/models/fastainn/tabular_nn_fastai.py

```diff
@@ -0,0 +1,334 @@
+
+            'recall_macro': 'recall',
+            'recall_micro': 'recall',
+            'recall_weigthed': 'recall',
+            'log_loss': 'valid_loss',
+        }
+        objective_func_name_to_monitor = objective_func_name
+        if objective_func_name in monitor_obj_func:
+            objective_func_name_to_monitor = monitor_obj_func[objective_func_name]
+        return objective_func_name_to_monitor
+
+    def predict_proba(self, X, preprocess=True):
+        from fastai.basic_data import DatasetType
+        from fastai.tabular import TabularList
+        from fastai.utils.mod_display import progress_disabled_ctx
+        from fastai.tabular import FillMissing, Categorify, Normalize
+
+        if preprocess:
+            X = self.preprocess(X)
+        procs = [FillMissing, Categorify, Normalize]
+        self.model.data.add_test(TabularList.from_df(X, cat_names=self.cat_columns, cont_names=self.cont_columns, procs=procs))
+        with progress_disabled_ctx(self.model) as model:
+            preds, _ = model.get_preds(ds_type=DatasetType.Test)
+        if self.problem_type == REGRESSION:
+            if self.y_scaler is not None:
+                return self.y_scaler.inverse_transform(preds.numpy()).reshape(-1)
+            else:
+                return preds.numpy().reshape(-1)
+        if self.problem_type == BINARY:
+            return preds[:, 1].numpy()
+        else:
+            return preds.numpy()
+
+    def save(self, file_prefix='', directory=None, return_filename=False, verbose=True):
+        # Export model
+        save_pkl.save_with_fn(f'{self.path}{self.model_internals_file_name}', self.model, lambda m, buffer: m.export(buffer, destroy=True), verbose=verbose)
+        self.model = {}
+        return super().save(file_prefix=file_prefix, directory=directory, return_filename=return_filename, verbose=verbose)
+
+    @classmethod
+    def load(cls, path: str, file_prefix='', reset_paths=False, verbose=True):
+        from fastai.basic_train import load_learner
+        obj = super().load(path, file_prefix=file_prefix, reset_paths=reset_paths, verbose=verbose)
+        obj.model = load_pkl.load_with_fn(f'{obj.path}{obj.model_internals_file_name}', lambda p: load_learner(obj.path, p), verbose=verbose)
+        return obj
+
+    def _set_default_params(self):
+        """ Specifies hyperparameter values to use by default """
+        default_params = get_param_baseline(self.problem_type)
+        for param, val in default_params.items():
+            self._set_default_param_value(param, val)
+
+    def _get_default_searchspace(self):
+        return get_default_searchspace(self.problem_type, num_classes=None)
+
+    def hyperparameter_tune(self, X_train, X_test, Y_train, Y_test, scheduler_options=None, **kwargs):
+        # TODO: add warning regarding dataloader leak: https://github.com/pytorch/pytorch/issues/31867
+        # TODO that hyperparameter-tuning is not yet implemented
```

Comment:
Will this leak make HPO infeasible for this model?

Reply:
This leak is visible only on hundreds of runs, so I don't think it will significantly impact HPO process here.